nncf/quantization/algorithms/accuracy_control/ranker.py

```diff
@@ -228,6 +228,31 @@ def _multiprocessing_calculation_ranking_score(
 
         return ranking_scores
 
+    def _multithreading_calculation_ranking_score(
+        self,
+        quantized_model: TModel,
+        quantized_model_graph: NNCFGraph,
+        groups_to_rank: List[GroupToRank],
+        ranking_subset_indices: List[int],
+    ):
+        
+        ranking_scores = []  # ranking_scores[i] is the ranking score for groups_to_rank[i]
+        modified_models = []
+        for current_group in groups_to_rank:
+            modified_model = revert_operations_to_floating_point_precision(
+                current_group.operations, current_group.quantizers, quantized_model, quantized_model_graph
+            )
+
+            modified_models.append(modified_model)
```

Comment:
A general comment is that the proposed solution is not memory optimal. What I mean is that this solution requires storing as many copies of the model in memory as there are groups in groups_to_rank. For some models, the number of groups for ranking is in the hundreds. Probably, algorithm can be crashed by memory for huge models.